DEVELOPMENT.md

```diff
@@ -0,0 +1,318 @@
+<!-- spell-checker:ignore reimplementing toybox RUNTEST CARGOFLAGS nextest -->
+
+# Setting up local development environment
+
+For contributing rules and best practices please refer to [CONTRIBUTING.md](CONTRIBUTING.md)
+
+## Before you start
+
+For this guide we assume that you already have GitHub account and have `git` and your favorite code editor or IDE installed and configured to work with GitHub.
+Before you start working on coreutils, please follow these steps:
+
+1. Fork [coreutils repository](https://github.com/uutils/coreutils) to your GitHub account [ref:](https://docs.github.com/en/get-started/quickstart/fork-a-repo)
+2. Clone that fork to your local development environment:
+```shell
+git clone https://github.com/YOUR-GITHUB-ACCOUNT/coreutils
+cd coreutils
+```
+
+## Tools
+
+You will need the tools mentioned in this section to build and test your code changes locally.
+This section will explain how to install and configure these tools.
+We also have an extensive CI that uses these tools and will check your code before it can be merged. 
+The next section [Testing](##Testing) will explain how to run those checks locally to avoid waiting for the CI.
+
+### Rust and friends
+
+Install [Rust](https://www.rust-lang.org/tools/install)
+
+If you're using rustup to install and manage your Rust toolchains, `cargo`, `clippy` and `rustfmt` are usually already installed. 
+You might also need to add 'llvm-tools':
+`rustup component add llvm-tools-preview` 
+
+**Tip:On MacOS** you will need to install C compiler & linker:
+`xcode-select --install`
+
+**Tip:On Windows** On Windows you'll need the MSVC build tools for Visual Studio 2013 or later.
+To acquire the build tools, you’ll need to install Visual Studio 2022. When asked which workloads to install, include:
+- Desktop Development with C++
+- The Windows 10 or 11 SDK
+- The English language pack component, along with any other language pack of your choosing
```

Comment:
Do we need to mention all of this? If it's a very common pitfall for newcomers I'm all for including it, but it feels maybe a bit too detailed? All bits of information here also need to be kept up to date (i.e. when people should install Visual Studio 2023 instead of 2022). So, just saying "install rust via rustup" should probably be fine. We could also link to more extensive installation guides instead.

This is how simple it could be: https://github.com/nushell/nushell/blob/main/CONTRIBUTING.md#setup (though we could add a bit more info if we really want to).

Reply:
True, could be simplified. On the Nu shell example though - they offload all details to linked https://www.nushell.sh/book/installation.html for installing Rust toolchain and all dependencies/prerequisites - that makse their CONTRIBUTING.md clean and simple. Woulkd the equivalent in our case be moving all those details out of CONTRIBUTING.md to DEVELOPMENT.md ?

Reply:
Hmm, yeah I guess you're right. We can keep it and just trim it a little bit.